### DISCUSSION TITLE
LOTW certificate expiry reminder?

### DISCUSSION THREAD

**Discussion by timkirby** [magicbug/Cloudlog#1860] [Ideas / Feature Requests]
I wondered why my recent QSOs hadn’t been uploaded to LOTW. Well, the reason was simple, my certificate had expired. 

Because Cloudlog is so good at uploading records, it’s easy to miss that a certificate is expiring. As Cloudlog knows the date a certificate is expiring, could there be a warning if the date is getting close?

**Comment by magicbug** [comment]
Reasonable suggestion, 30 days?

**Reply by magicbug** [reply]
Then at 5 days show a constant reminder on the dashboard when you login

**Comment by timkirby** [comment]
Sounds great. Maybe could be in the banner at the top of the dashboard where it says ‘you have had x QSOs today’? Append a warning like ‘check for expiring LOTW certificates’ or similar?On 22 Dec 2022, at 22:28, Peter Goodhall ***@***.***> wrote:﻿
Reasonable suggestion, 30 days?

**Comment by timkirby** [comment]
That would work too!On 22 Dec 2022, at 22:30, Peter Goodhall ***@***.***> wrote:﻿
Then at 5 days show a constant reminder on the dashboard when you login

**Comment by phl0** [comment]
I started some work and added a warning notifier to the LotW certs that shows an orange icon 10 days before expiry. 

![Screenshot from 2022-12-23 10-57-12](https://user-images.githubusercontent.com/7112907/209314970-ec3a4b7c-fed9-4df9-bb15-8e3559eaaca8.png)


Code is here: https://github.com/phl0/Cloudlog/tree/lotwWarning

Will also add a notification to the dashboard.

**Comment by timkirby** [comment]
That looks great, Flo - thank you for the speedy response!

**Reply by phl0** [reply]
> That looks great, Flo - thank you for the speedy response!

No worries. This is not what you wished for but as I was at it discovering (again) how the code works here I thought this might also be helpful. Guess I will also make the 10 day more like 30. Opinions?

**Comment by timkirby** [comment]
30 is probably good. It looks like ARRL may take 3-4 business days to respond - so it’s definitely better not to leave it to the wire!

**Comment by phl0** [comment]
So my patch is complete now. It checks for expired certificates as well as certificates expiring in the next 30 days:

![Screenshot from 2022-12-23 14-47-46](https://user-images.githubusercontent.com/7112907/209347683-9979aa4b-de84-4887-be97-1fe298629846.png)

Are you able to test this?

**Comment by timkirby** [comment]
Excellent! I don’t know whether Pete is able to push the code change to my hosted instance. I’ve got my replacement certificate, but I can hold of installing it until I’ve tested it.

**Comment by phl0** [comment]
Ah then you are on master branch most probably. This goes to dev first. So I may ask @magicbug or @AndreasK79 for a review before merging it :)

**Comment by timkirby** [comment]
Yes, I’m almost certainly on the master branch! Thanks again, Flo!
